creativecloud/blocks/interactive-metadata/interactive-metadata.js

```diff
@@ -0,0 +1,294 @@
+import { getLibs } from '../../scripts/utils.js';
+import defineDeviceByScreenSize from '../../scripts/decorate.js';
+
+function getImgSrc(pic) {
+  const viewport = defineDeviceByScreenSize() === 'MOBILE' ? 'mobile' : 'desktop';
+  let source = '';
+  if (viewport === 'mobile') source = pic.querySelector('source[type="image/webp"]:not([media])');
+  else source = pic.querySelector('source[type="image/webp"][media]');
+  return source.srcset;
+}
+
+function getNextStepIndex(stepInfo) {
+  return (stepInfo.stepIndex + 1) % stepInfo.stepList.length;
+}
+
+function getPrevStepIndex(stepInfo) {
+  return stepInfo.stepIndex - 1 >= 0
+    ? stepInfo.stepIndex - 1
+    : stepInfo.stepList.length - 1;
+}
+
+async function loadJSandCSS(stepName) {
+  const miloLibs = getLibs('/libs');
+  const { loadStyle } = await import(`${miloLibs}/utils/utils.js`);
+  const stepJS = `${window.location.origin}/creativecloud/features/interactive-components/${stepName}/${stepName}.js`;
+  const stepCSS = `${window.location.origin}/creativecloud/features/interactive-components/${stepName}/${stepName}.css`;
+  loadStyle(stepCSS);
+  const { default: initFunc } = await import(stepJS);
+  return initFunc;
+}
+
+function loadImg(img) {
+  if (!img) return;
+  return new Promise((res) => {
+    img.loading = 'eager';
+    if (img.complete) {
+      res();
+      return;
+    }
+    img.onload = () => res();
```

Comment:
Can we try img.onload = res, if it works.

Reply:
I tried but then it resolves immediately hence added the arrow function

Reply:
> Can we try img.onload = res, if it works.

I tried but then it resolves immediately hence added the arrow function